src/journal/journald-syslog.c

```diff
@@ -326,26 +326,35 @@ void server_process_syslog_message(
                         log_warning_errno(r, "Failed to retrieve credentials for PID " PID_FMT ", ignoring: %m", ucred->pid);
         }
 
-        orig = buf;
-        syslog_parse_priority(&buf, &priority, true);
+        /* We are creating copy of the message because we want to forward original message verbatim to the legacy
+           syslog implementation */
+        for (i = buf_len; i > 0; i--)
+                if (!strchr(WHITESPACE, buf[i-1]))
+                        break;
+
+        msg = newa0(char, i + 1);
+        memcpy(msg, buf, i);
+        msg += strspn(msg, WHITESPACE);
```

Comment:
msg = skip_leading_chars(msg, WHITESPACE) is a tiny bit easier to read... but doesn't matter